src/many-ledger/src/storage.rs

```diff
@@ -939,6 +972,130 @@ impl LedgerStorage {
             ..Default::default()
         })
     }
+
+    // IdStore
+    pub fn store(
+        &mut self,
+        recall_phrase: &RecallPhrase,
+        address: &Identity,
+        cred_id: CredentialId,
+        public_key: PublicKey,
+    ) -> Result<(), ManyError> {
+        let recall_phrase_cbor = minicbor::to_vec(recall_phrase)
+            .map_err(|e| ManyError::serialization_error(e.to_string()))?;
+        if self
+            .persistent_store
+            .get(&recall_phrase_cbor)
+            .map_err(|e| ManyError::unknown(e.to_string()))?
+            .is_some()
+        {
+            return Err(idstore::existing_entry());
+        }
+        let address_vec = address.to_vec();
+        let value = minicbor::to_vec(vec![
+            Vec::<u8>::from(cred_id.0),
+            Vec::<u8>::from(public_key.0),
+        ])
+        .map_err(|e| ManyError::serialization_error(e.to_string()))?;
+
+        // Keys in batch must be sorted.
+        let batch = match recall_phrase_cbor.cmp(&address_vec) {
+            Ordering::Less | Ordering::Equal => vec![
+                (
+                    vec![IDSTORE_ROOT, &recall_phrase_cbor].concat(),
```

Comment:
If you add a prefix to recall_phrase_cbor and address here you don't have to compare.